templates/base/generators/hooks/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Hooks Generator
+ */
+
+/* eslint strict: ["off"] */
+
+'use strict';
+
+const componentExists = require('../utils/componentExists');
+const config = require('../constants');
+
+module.exports = {
+  description: 'Add an hook',
+  prompts: [
+    {
+      type: 'input',
+      name: 'name',
+      message: 'What should it be called?',
+      default: 'ChangeTitle',
+      validate: (value) => {
+        value = `use${value}`;
+        if (/.+/.test(value)) {
+          return componentExists(value, 'hooks')
+            ? 'A hook with this name already exists '
+            : true;
+        }
+
+        return 'The name is required';
+      }
+    }
+  ],
+  actions: (data) => {
+    // Generate useHookName.js and useHookName.test.js
+    const folderPath = `../${config.HOOKS_PATH}`;
+    const actions = [
+      {
+        type: 'add',
+        path: `${folderPath}/use{{properCase name}}.js`,
+        templateFile: './hooks/hook.js.hbs',
+        abortOnFail: true
+      },
+      {
+        type: 'add',
+        path: `${folderPath}/use{{properCase name}}.test.js`,
```

Comment:
@pras9 please make following adjustment 

for each hook that gets created under src/hooks should have a directory by same name. Such that  src/hooks/useChangeTitle/

Add a index.js file similar to what we have in component that will export the hook out easily

Reply:
PR updated with the changes.